pyrealm_build_data/splash/splash_py_version/splash_data.py

```diff
@@ -0,0 +1,834 @@
+        self.data_dir["tmp"] = d
+
+    def set_watch_rainf_dir(self, d):
+        """
+        Name:     SPLASH_DATA.set_watch_rainf_dir
+        Input:    str, directory path (d)
+        Features: Define the directory to WATCH daily rainfall netCDF file
+        """
+        logging.info("WATCH precipitation directory set to %s", d)
+        self.data_dir["Rainf"] = d
+
+    def set_watch_tair_dir(self, d):
+        """
+        Name:     SPLASH_DATA.set_watch_tair_dir
+        Input:    str, directory path (d)
+        Features: Define the directory to WATCH daily air temperature netCDF
+                  file
+        """
+        logging.info("WATCH temperature directory set to %s", d)
+        self.data_dir["Tair"] = d
+
+    def set_output_dir(self, d):
+        """
+        Name:     SPLASH_DATA.set_output_dir
+        Input:    str, directory path (d)
+        Features: Define the directory where the output data will be written
+                  and create the output file
+        Depends:  writeout
+        """
+        self.output_file = d + "splash_data_out.csv"
+        header = "date,sf,tair,pn\n"
+        if os.path.isfile(self.output_file):
+            try:
+                user_resp = raw_input("File already exists, overwrite (y/n)? ")
+            except NameError:
+                # Maybe using Python 3, try again:
+                try:
+                    user_resp = input("File already exists, overwrite (y/n)? ")
+                except:
+                    logging.exception("Could not get user input")
+                    raise
+            if user_resp.lower() == "y":
+                self.writeout(self.output_file, header)
+        else:
+            self.writeout(self.output_file, header)
+            logging.info("creating output file %s", self.output_file)
+
+    def set_sf_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_sf_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for fractional sunshine hours
+                  (e.g., 'cru' or 'watch')
+        """
+        logging.info("setting sunshine fraction preference to '%s'", n)
+        self.data_src["sf"] = n
+
+    def set_pn_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_pn_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for precipitation (e.g., 'cru'
+                  or 'watch')
+        """
+        logging.info("setting precipitation preference to '%s'", n)
+        self.data_src["pn"] = n
+
+    def set_tair_source(self, n):
+        """
+        Name:     SPLASH_DATA.set_tair_source
+        Input:    str, source preference (n)
+        Features: Define the source preference for air temperature (e.g., 'cru'
+                  or 'watch')
+        """
+        logging.info("setting temperature preference to '%s'", n)
+        self.data_src["tair"] = n
+
+    def add_one_month(self, dt0):
+        """
```

Comment:
We can mention the necessity to add/sub a month to the datetime.

Reply:
@surbhigoel77 - Don't review the code files in `splash_py_version` - there's all sorts of oddities in the code, but its the benchmark implementation so I'm leaving them as is.